### DISCUSSION TITLE
How to work with gd.Callable?

### DISCUSSION THREAD

**Discussion by ullanar** [quaadgras/graphics.gd#51] [Q&A] [answered]
Hello. Currently working on `InteractionManager` and process involves me into `gd.Array` type. 
Idea was to get all overlapping areas each frame, checks if there are any specific "`Interactive`" areas and if so then show player what he can interact with related entity

So there are several `gd.ArrayOf` methods which requires `gd.Callable` as argument. I +- understand how it works in gdscript but here I cant do almost nothing x)

So how can I properly create `gd.Callable` and implement some logic into?

```go
	overlappingAreas := i.Area.GetOverlappingAreas(i.Temporary)
	overlappingAreas.Filter(i.Temporary, method gd.Callable)

```

**Comment by Splizard** [comment, marked as answer]
You can create callables out of any Go function, ie. `Callable.New(func() { fmt.Println("called from godot") })`, these can take arguments too (and return a value) but you will need to know which type of arguments Godot will be passing in advance.


```go
filter := Callable.New(func(area Area2D.Instance) bool {
    return ...
})
